diff --git a/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js b/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
index f857a06..cd9778f 100644
--- a/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
+++ b/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
@@ -326,7 +326,9 @@ class Whatsapp {
                     auth: store.state,
                     logger: this.P,
                     markOnlineOnConnect: false,
-                    browser: baileys_1.Browsers.ubuntu("Chrome"),
+                    syncFullHistory: false,
+                    generateHighQualityLinkPreview: false,
+                    browser: baileys_1.Browsers.macOS("Desktop"),
                 });
                 this.sessions.set(sessionId, {
                     sock: sock,
@@ -448,7 +450,9 @@ class Whatsapp {
                     auth: store.state,
                     logger: this.P,
                     markOnlineOnConnect: false,
-                    browser: baileys_1.Browsers.ubuntu("Chrome"),
+                    syncFullHistory: false,
+                    generateHighQualityLinkPreview: false,
+                    browser: baileys_1.Browsers.macOS("Desktop"),
                 });
                 this.sessions.set(sessionId, {
                     sock: sock,
