diff --git a/node_modules/wa-multi-session/dist/Types/Whatsapp.d.ts b/node_modules/wa-multi-session/dist/Types/Whatsapp.d.ts
index 05767f3..c1885d5 100644
--- a/node_modules/wa-multi-session/dist/Types/Whatsapp.d.ts
+++ b/node_modules/wa-multi-session/dist/Types/Whatsapp.d.ts
@@ -10,7 +10,7 @@ export type WhatsappConstructorProps = {
      */
     onConnecting?: (sessionId: string) => any;
     onConnected?: (sessionId: string) => any;
-    onDisconnected?: (sessionId: string) => any;
+    onDisconnected?: (sessionId: string, code?: number, error?: any, shouldRetry?: boolean, retryAttempt?: number) => any;
     onPairingCode?: (sessionId: string, code: string) => void;
     onMessageUpdated?: (data: MessageUpdated) => any;
     onMessageReceived?: (msg: MessageReceived) => any;
diff --git a/node_modules/wa-multi-session/dist/Types/index.d.ts b/node_modules/wa-multi-session/dist/Types/index.d.ts
index 9f95c3f..98d652e 100644
--- a/node_modules/wa-multi-session/dist/Types/index.d.ts
+++ b/node_modules/wa-multi-session/dist/Types/index.d.ts
@@ -58,7 +58,7 @@ export interface StartSessionParams {
     onQRUpdated?: (qr: string) => void;
     onConnected?: () => void;
     onConnecting?: () => void;
-    onDisconnected?: () => void;
+    onDisconnected?: (code?: number, error?: any, shouldRetry?: boolean, retryAttempt?: number) => void;
     onMessageReceived?: (message: MessageReceived) => void;
     onMessageUpdated?: (message: MessageUpdated) => void;
     store?: LegacyStore;
@@ -71,7 +71,7 @@ export interface StartSessionWithPairingCodeParams {
     onQRUpdated?: (qr: string) => void;
     onConnected?: () => void;
     onConnecting?: () => void;
-    onDisconnected?: () => void;
+    onDisconnected?: (code?: number, error?: any, shouldRetry?: boolean, retryAttempt?: number) => void;
     onPairingCode?: (code: string) => void;
     onMessageReceived?: (message: MessageReceived) => void;
     onMessageUpdated?: (message: MessageUpdated) => void;
diff --git a/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js b/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
index f857a06..3623610 100644
--- a/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
+++ b/node_modules/wa-multi-session/dist/Whatsapp/Whatsapp.js
@@ -322,11 +322,13 @@ class Whatsapp {
             const startSocket = () => __awaiter(this, void 0, void 0, function* () {
                 const store = yield this.getStore(sessionId);
                 const sock = (0, baileys_1.default)({
-                    version,
+                    version: [2, 3000, 1015901307],
                     auth: store.state,
                     logger: this.P,
                     markOnlineOnConnect: false,
-                    browser: baileys_1.Browsers.ubuntu("Chrome"),
+                    syncFullHistory: false,
+                    generateHighQualityLinkPreview: false,
+                    browser: ['Mac OS', 'Chrome', '121.0.6167.159'],
                 });
                 this.sessions.set(sessionId, {
                     sock: sock,
@@ -360,7 +362,9 @@ class Whatsapp {
                                     this.sessions.get(sessionId).status = "connecting";
                             }
                             if (connection === "close") {
-                                const code = (_f = (_e = lastDisconnect === null || lastDisconnect === void 0 ? void 0 : lastDisconnect.error) === null || _e === void 0 ? void 0 : _e.output) === null || _f === void 0 ? void 0 : _f.statusCode;
+                                const errorObj = (_e = lastDisconnect === null || lastDisconnect === void 0 ? void 0 : lastDisconnect.error) === null || _e === void 0 ? void 0 : _e.output;
+                                const code = (_f = errorObj) === null || _f === void 0 ? void 0 : _f.statusCode;
+                                console.log("[baileys-debug] connection close. Code:", code, "Payload:", errorObj?.payload);
                                 let retryAttempt = (_g = this.retryCount.get(sessionId)) !== null && _g !== void 0 ? _g : 0;
                                 let shouldRetry = false;
                                 if (code != baileys_1.DisconnectReason.loggedOut && retryAttempt < 10) {
@@ -368,7 +372,10 @@ class Whatsapp {
                                 }
                                 if (shouldRetry) {
                                     retryAttempt++;
+                                    console.log("[baileys-debug] Retrying connection...", retryAttempt);
                                     this.retryCount.set(sessionId, retryAttempt);
+                                    (_h = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _h === void 0 ? void 0 : _h(sessionId, code, errorObj, true, retryAttempt);
+                                    (_j = options.onDisconnected) === null || _j === void 0 ? void 0 : _j.call(options, code, errorObj, true, retryAttempt);
                                     startSocket();
                                 }
                                 else {
@@ -377,8 +384,8 @@ class Whatsapp {
                                         this.sessions.get(sessionId).status = "disconnected";
                                     this.retryCount.delete(sessionId);
                                     this.deleteSession(sessionId);
-                                    (_h = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _h === void 0 ? void 0 : _h(sessionId);
-                                    (_j = options.onDisconnected) === null || _j === void 0 ? void 0 : _j.call(options);
+                                    (_h = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _h === void 0 ? void 0 : _h(sessionId, code, errorObj, false, retryAttempt);
+                                    (_j = options.onDisconnected) === null || _j === void 0 ? void 0 : _j.call(options, code, errorObj, false, retryAttempt);
                                 }
                             }
                             if (connection == "open") {
@@ -444,11 +451,13 @@ class Whatsapp {
                 let isPairingCodeRequested = false;
                 const store = yield this.getStore(sessionId);
                 const sock = (0, baileys_1.default)({
-                    version,
+                    version: [2, 3000, 1015901307],
                     auth: store.state,
                     logger: this.P,
                     markOnlineOnConnect: false,
-                    browser: baileys_1.Browsers.ubuntu("Chrome"),
+                    syncFullHistory: false,
+                    generateHighQualityLinkPreview: false,
+                    browser: baileys_1.Browsers.macOS("Desktop"),
                 });
                 this.sessions.set(sessionId, {
                     sock: sock,
@@ -492,7 +501,9 @@ class Whatsapp {
                                     this.sessions.get(sessionId).status = "connecting";
                             }
                             if (connection === "close") {
-                                const code = (_g = (_f = lastDisconnect === null || lastDisconnect === void 0 ? void 0 : lastDisconnect.error) === null || _f === void 0 ? void 0 : _f.output) === null || _g === void 0 ? void 0 : _g.statusCode;
+                                const errorObj = (_f = lastDisconnect === null || lastDisconnect === void 0 ? void 0 : lastDisconnect.error) === null || _f === void 0 ? void 0 : _f.output;
+                                const code = (_g = errorObj) === null || _g === void 0 ? void 0 : _g.statusCode;
+                                console.log("[baileys-debug] connection close. Code:", code, "Payload:", errorObj?.payload);
                                 let retryAttempt = (_h = this.retryCount.get(sessionId)) !== null && _h !== void 0 ? _h : 0;
                                 let shouldRetry = false;
                                 if (code != baileys_1.DisconnectReason.loggedOut && retryAttempt < 10) {
@@ -500,7 +511,10 @@ class Whatsapp {
                                 }
                                 if (shouldRetry) {
                                     retryAttempt++;
+                                    console.log("[baileys-debug] Retrying connection...", retryAttempt);
                                     this.retryCount.set(sessionId, retryAttempt);
+                                    (_j = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _j === void 0 ? void 0 : _j(sessionId, code, errorObj, true, retryAttempt);
+                                    (_k = options.onDisconnected) === null || _k === void 0 ? void 0 : _k.call(options, code, errorObj, true, retryAttempt);
                                     startSocket();
                                 }
                                 else {
@@ -509,8 +523,8 @@ class Whatsapp {
                                         this.sessions.get(sessionId).status = "disconnected";
                                     this.retryCount.delete(sessionId);
                                     this.deleteSession(sessionId);
-                                    (_j = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _j === void 0 ? void 0 : _j(sessionId);
-                                    (_k = options.onDisconnected) === null || _k === void 0 ? void 0 : _k.call(options);
+                                    (_j = this.callback.get(Defaults_1.CALLBACK_KEY.ON_DISCONNECTED)) === null || _j === void 0 ? void 0 : _j(sessionId, code, errorObj, false, retryAttempt);
+                                    (_k = options.onDisconnected) === null || _k === void 0 ? void 0 : _k.call(options, code, errorObj, false, retryAttempt);
                                 }
                             }
                             if (connection == "open") {
